python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -628,49 +400,196 @@ class RandomForestClassifier(Base):
+
+        if self.dtype == np.float32:
+            fit(handle_[0],
+                rf_forest,
+                <float*> X_ptr,
+                <int> n_rows,
+                <int> self.n_cols,
+                <int*> y_ptr,
+                <int> num_unique_labels,
+                rf_params)
+
+        else:
+            fit(handle_[0],
+                rf_forest64,
+                <double*> X_ptr,
+                <int> n_rows,
+                <int> self.n_cols,
+                <int*> y_ptr,
+                <int> num_unique_labels,
+                rf_params64)
+        # make sure that the `fit` is complete before the following delete
+        # call happens
+        self.handle.sync()
+        del(X_m)
+        del(y_m)
+        return self
 
     def predict(self, X):
-        """
-        Predicts the labels for X.
 
-        Parameters
-        ----------
-        X : array-like (host) shape = (n_samples, n_features)
-            Dense matrix (floats or doubles) of shape (n_samples, n_features).
-            Acceptable formats: NumPy ndarray, Numba device
-            ndarray
+        cdef uintptr_t X_ptr
+        # row major format
+        X_m, X_ptr, n_rows, n_cols, _ = \
+            input_to_dev_array(X, order='C')
+        if n_cols != self.n_cols:
+            raise ValueError("The number of columns/features in the training"
+                             " and test data should be the same ")
+        if X.dtype != self.dtype:
+            raise ValueError("The datatype of the training data is different"
+                             " from the datatype of the testing data")
 
-        Returns
-        ----------
-        y: NumPy
-           Dense vector (int) of shape (n_samples, 1)
+        preds = np.zeros(n_rows, dtype=np.int32)
+        cdef uintptr_t preds_ptr
+        preds_m, preds_ptr, _, _, _ = \
+            input_to_dev_array(preds)
+        cdef cumlHandle* handle_ =\
+            <cumlHandle*><size_t>self.handle.getHandle()
 
-        """
+        cdef RandomForestMetaData[float, int] *rf_forest = \
+            <RandomForestMetaData[float, int]*><size_t> self.rf_forest
+
+        cdef RandomForestMetaData[double, int] *rf_forest64 = \
+            <RandomForestMetaData[double, int]*><size_t> self.rf_forest64
+        if self.dtype == np.float32:
+            predict(handle_[0],
+                    rf_forest,
+                    <float*> X_ptr,
+                    <int> n_rows,
+                    <int> n_cols,
+                    <int*> preds_ptr,
+                    <bool> self.verbose)
 
-        return self._impl.predict(X)
+        elif self.dtype == np.float64:
+            predict(handle_[0],
+                    rf_forest64,
+                    <double*> X_ptr,
+                    <int> n_rows,
+                    <int> n_cols,
+                    <int*> preds_ptr,
+                    <bool> self.verbose)
+        else:
+            raise TypeError("supports only float32 and float64 input,"
```

Comment:
```suggestion
            raise TypeError("supports only np.float32 and np.float64 input,"
```

Either all dtypes are using np.* nomenclature or not for consistency, something we should improve in all our messaging (should be dealt with once the input dtype PR is done eventually)